### ISSUE TITLE
Suggestion to raise recursion limit is missing

### ISSUE THREAD

**Issue by bluss** [rust-lang/rust#36844]
Rust used to suggest the attribute that raises the recursion limit when this error was triggered. It does not suggest an attribute now.

``` rust
error: recursion limit reached while expanding the macro `parse_macros_util`
   --> src/lib.rs:366:5
    |
366 |     custom_derive! {
    |     ^ in this macro invocation
    |
    = note: this error originates in a macro outside of the current crate

error: Could not compile `rand_derive`.
```

encountered with: rustc 1.13.0-nightly (d0623cf7b 2016-09-26)

**Comment by Mark-Simulacrum** [comment]
I believe this is fixed (or at least, I don't see an example here that I could verify with and my attempt worked).

```rust
macro_rules! a {
    () => { a!() }
}

fn main() {
    a!()
}
```

produces

```
error: recursion limit reached while expanding the macro `a`
 --> test.rs:2:13
  |
2 |     () => { a!() }
  |             ^^^^
...
6 |     a!()
  |     ---- in this macro invocation
  |
  = help: consider adding a `#![recursion_limit="128"]` attribute to your crate
```
